### PR TITLE
Integrate Recents tab in Data Picker

### LINKED ISSUE #42675: Integrate Recents tab
- [x] enable recents tab in the Data Picker
- [x] update existing e2e tests to account for the new Recents tab (which will now be the default tab)
- [x] add new e2e tests to account for the new Recents tab

### PR DESCRIPTION
**Pull request by kamilmielnik**
Closes #42675

### Description

Enables the "Recents" tab in the Data Picker.
Updates few existing assertions and adds new ones to account for the new tab.
`notebook-data-source.cy.spec.ts` is the main suite for the Data Picker.

### How to verify

Ci is all green.
Recents tab is enabled in the Data Picker and Join Data Picker.

### CODE REVIEW

File: frontend/src/metabase/query_builder/components/notebook/steps/DataStep/DataStep.unit.spec.tsx [hunk 1]
```diff
@@ -106,6 +109,9 @@ describe("DataStep", () => {
   });
 
   afterAll(() => {
+    HTMLElement.prototype.scrollBy = scrollBy;
+    HTMLElement.prototype.getBoundingClientRect = getBoundingClientRect;
```
Comment (kamilmielnik) [inline frontend/src/metabase/query_builder/components/notebook/steps/DataStep/DataStep.unit.spec.tsx:113]: Cleaning up the mocks, addressing feedback from previous PR: https://github.com/metabase/metabase/pull/42565/files#r1604585537

File: frontend/src/metabase/common/components/DataPicker/components/DataPickerModal.tsx [hunk 2]
```diff
@@ -49,7 +49,7 @@ const options: DataPickerModalOptions = {
   hasConfirmButtons: false,
   showPersonalCollections: true,
   showRootCollection: true,
-  hasRecents: false, // TODO: https://github.com/metabase/metabase/issues/42675
+  hasRecents: true,
```
Comment (kamilmielnik) [inline frontend/src/metabase/common/components/DataPicker/components/DataPickerModal.tsx:52]: Changes in this file enable the Recents tab

File: e2e/test/scenarios/question/notebook-data-source.cy.spec.ts [hunk 3]
```diff
@@ -272,6 +289,12 @@ describe("scenarios > notebook > data source", () => {
         assertDataPickerEntitySelected(1, "First collection");
         assertDataPickerEntitySelected(2, "Second collection");
         assertDataPickerEntitySelected(3, checkNotNull(modelDetails.name));
+
+        entityPickerModalTab("Recents").click();
+        cy.findByTestId("result-item")
+          .should("exist")
+          .and("contain.text", checkNotNull(modelDetails.name))
+          .and("have.css", "background-color", "rgb(80, 158, 227)");
```
Comment (uladzimirdev) [inline e2e/test/scenarios/question/notebook-data-source.cy.spec.ts:297]: @kamilmielnik why do we want to assert css?

**Review by uladzimirdev — APPROVED** [review]
LGTM

File: e2e/test/scenarios/question/notebook-data-source.cy.spec.ts (hunk 3 above)
Comment (kamilmielnik) [reply e2e/test/scenarios/question/notebook-data-source.cy.spec.ts:297]: Currently that's the only way to determine if a given item is selected or not.

Alternatively we could add `aria-selected` attribute. WDYT?

Comment (uladzimirdev) [reply e2e/test/scenarios/question/notebook-data-source.cy.spec.ts:297]: not critial, but worth having a comment in the code

`aria-selected` or `aria-active` work for me

Comment (kamilmielnik) [reply e2e/test/scenarios/question/notebook-data-source.cy.spec.ts:297]: Addressed here: #42825

**Pull request merged**